client.go

```diff
@@ -54,13 +63,71 @@ func (a *authCreds) RequireTransportSecurity() bool {
 	return true
 }
 
+// Open creates a new Dgraph client by parsing a connection string of the form:
+// dgraph://<optional-login>:<optional-password>@<host>:<port>?<optional-params>
+// For example `dgraph://localhost:9080?sslmode=require`
+// It connects to the gRPC endpoint and, if credentials are provided, signs in the user.
+// An error can be returned if the Dgraph cluster is not yet ready to accept requests--the text
+// of the error in this case will contain the string "Please retry".
+func Open(connStr string) (*Dgraph, error) {
+	u, err := url.Parse(connStr)
+	if err != nil {
+		return nil, fmt.Errorf("invalid connection string: %w", err)
+	}
+
+	if u.Scheme != dgraphScheme {
+		return nil, fmt.Errorf("invalid scheme: must start with %s://", dgraphScheme)
+	}
+
+	opts := []ClientOption{}
+
+	apiKey := u.Query().Get(cloudAPIKeyParam)
+	sslMode := u.Query().Get(sslModeParam)
+
+	if apiKey != "" {
+		opts = append(opts, WithDgraphAPIKey(apiKey))
+	}
+
+	if sslMode == "" {
+		sslMode = sslModeDisable
+	}
+	switch sslMode {
+	case sslModeDisable:
+		opts = append(opts, WithGrpcOption(grpc.WithTransportCredentials(insecure.NewCredentials())))
+	case sslModeRequire:
+		opts = append(opts, WithSkipTLSVerify())
+	case sslModeVerifyCA:
+		opts = append(opts, WithSystemCertPool())
+	default:
+		return nil, fmt.Errorf("invalid SSL mode: %s (must be one of %s, %s, %s)", sslMode, sslModeDisable, sslModeRequire, sslModeVerifyCA)
+	}
+
+	if u.User != nil {
```

Comment:
we should check for both here. If only one of them is present, we should let the user know that something is wrong.

Reply:
Here u.User is an object containing both username and password

Reply:
but what if I have provided the username and not provided the password? Will I go inside this if then?